src/lib/ui/SupportedDevices/FilterCategoryMenu.tsx

```diff
@@ -0,0 +1,63 @@
+import { ChevronDownIcon } from 'lib/icons/ChevronDown.js'
+import Menu from 'lib/ui/Menu/Menu.js'
+import { MenuItem } from 'lib/ui/Menu/MenuItem.js'
+
+interface FilterCategoryMenuProps {
+  label: string
+  options: string[]
+  addAllOption?: boolean
+  onSelect: (option: string) => void
+  onAllOptionSelect?: () => void
+  buttonLabel?: string
+}
+
+export function FilterCategoryMenu({
+  label,
+  options,
+  addAllOption,
+  onSelect,
+  onAllOptionSelect,
+  buttonLabel,
+}: FilterCategoryMenuProps) {
+  const usableOptions = addAllOption ? ['All', ...options] : options
+
+  if (Boolean(addAllOption) && onAllOptionSelect == null) {
+    throw new Error(
+      'onAllOptionSelect must be provided if addAllOption is true'
+    )
+  }
```

Comment:
We would like to avoid throwing errors in components like this. If the props have different modes we could represent this as a union in the prop type.